src/ReportManager/DetailsSection/SchemaForm/index.js

```diff
@@ -15,14 +15,38 @@ export const FIELDS = {
   [FORM_ELEMENT_TYPES.TEXT]: Text,
 };
 
-const SchemaForm = ({ formData, onFormChange, onFormSubmit, renderSubmitButton, schema }) => {
-  const fields = useMemo(() => schema ? makeFieldsFromSchema(schema) : {}, [schema]);
+const SchemaForm = ({ formData, onFormDataChange, onFormSubmit, renderSubmitButton, schema }) => {
+  const runValidations = useSchemaValidations(schema);
+
+  const [fieldErrors, setFieldErrors] = useState({});
+
+  const fields = useMemo(() => makeFieldsFromSchema(schema), [schema]);
+
+  const fieldValues = useMemo(() => Object.entries(formData).reduce((accumulator, [fieldId, fieldValue]) => {
+    // TODO: Collections will require recusivity here.
+
+    return { ...accumulator, [fieldId]: fieldValue };
+  }, {}), [formData]);
+
+  const onFieldChange = useCallback((fieldId, value) => {
+    // TODO: Collections will require recusivity here.
+    // If the value is empty, set it as undefined so AJV validation returns errors for required fields.
+    const newFormData = { ...formData, [fieldId]: value || undefined };
+
+    onFormDataChange(newFormData);
+    setFieldErrors((fieldErrors) => ({ ...fieldErrors, [fieldId]: undefined }));
+  }, [formData, onFormDataChange]);
 
   const onSubmit = (event) => {
     event.preventDefault();
 
-    // TODO: Add form validation with AJV before submission and show errors.
-    onFormSubmit({ formData });
+    const fieldErrors = runValidations(formData);
```

Comment:
**Related to AJV validations**
Before confirming the submission, we run the validations and show errors if there are any.